src/components/app.jsx

```diff
@@ -30,7 +36,17 @@ class App extends React.Component {
     }
   }
 
+  saveCards() {
+    localStorage.setItem('flash-cards', JSON.stringify(this.state.cards));
+  }
+
+  addCard(newCard) {
+    const newCards = this.state.cards.concat(newCard);
+    this.setState({ cards: newCards }, this.saveCards());
```

Comment:
Be careful not to call `saveCards` too soon. 

```js
    this.setState({ cards: newCards }, this.saveCards);
```
If you pass it like this (without calling it) it will be called for you at the right time. `setState` is asynchronous!